woocommerce/woocommerce
# Add To Cart With Options - Stepper layout: implement feature under feature flag

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



This PR hides the `Stepper Layout` feature behind an experimental flag to gather feedback before its official release. Additionally, it introduces utility functions to enable or disable the experimental flag in E2E tests.



Closes # .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Run `pnpm run --filter="@woocommerce/plugin-woocommerce" watch:build`.
2. Visit the Single Product Template.
3. Click on the Add To Cart With Options.
4. Ensure that the Quantity Selector setting is available.

![image](https://github.com/user-attachments/assets/c711a3b6-cbe0-45bf-81f8-5e9153a298f5)

5. Run `pnpm run --filter="@woocommerce/plugin-woocommerce" build:zip` (or install this zip: [woocommerce.zip](https://github.com/user-attachments/files/17649269/woocommerce.zip))
6. Visit the Single Product Template.
7. Click on the Add To Cart With Options.
8. Ensure that the Quantity Selector setting isn't available.
9. Install WooCommerce Beta Tester:
[woocommerce-beta-tester.zip](https://github.com/user-attachments/files/17629774/woocommerce-beta-tester.zip)
10. Visit `wp-admin/tools.php?page=woocommerce-admin-test-helper`.
11. Enable `add-to-cart-with-options-stepper-layout` feature.
12. Visit the Single Product Template.
13. Ensure that the Quantity Selector setting is available.




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Add To Cart With Options - Stepper layout: implement feature under feature flag

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
